### PR TITLE
fix: [PIE-7128]: Store error message on webhook registration when failure to fetch connector details

### PR DESCRIPTION
**Pull request by vcalasansh**
This PR makes the webhook registration flow for Git webhooks store (and show in the UI) the correct error message when there is a failure to fetch connector details. These connector details are used to register the webhook in Git provider's end.

When a user without `SECRET_ACCESS_PERMISSION` role (for one or more of the secrets used by the connector) was trying to save a git webhook trigger, the behavior was that webhook registration was failing silently, without any warning in the UI. Now we will show an instructive error message in the UI:

image
 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42121)

### CODE REVIEW

**Comment by vcalasansh** [comment]
trigger messagecheck

**Review by sribalij — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42121)* status: Some of the checks are too old. 1 file(s) to review. 1 of 2 approvals obtained.

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerWebhookRegistrationServiceImpl.java [hunk 1]
```diff
@@ -57,8 +57,20 @@ public WebhookRegistrationStatusData registerWebhook(NGTriggerEntity ngTriggerEn
...
+    ConnectorDetails connectorDetails;
+    try {
+      connectorDetails = connectorUtils.getConnectorDetails(
+          ngAccess, ngTriggerEntity.getMetadata().getWebhook().getGit().getConnectorIdentifier());
+    } catch (Exception ex) {
+      log.error("Failed to register webhook, could not fetch connector details", ex);
+      WebhookRegistrationStatusDataBuilder metadataBuilder = WebhookRegistrationStatusData.builder();
+      metadataBuilder.webhookAutoRegistrationStatus(
+          WebhookAutoRegistrationStatus.builder()
+              .detailedMessage("Failed to fetch connector details: " + ex.getMessage())
```
Comment (abhinav-harness) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerWebhookRegistrationServiceImpl.java:69]: ExceptionUtils.getMessage(ex)

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42121)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42121#-:-NKyK2SjBo20qePflzig:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**